### ISSUE TITLE
Vaccine course - Issue filling Age Range for doses

### ISSUE THREAD

**Issue by regotaina** [msupply-foundation/open-msupply#4884]
## What went wrong? 😲

When creating a new Vaccine Course, if you try to edit the age range starting with 'From age' the field won't accept any number smaller than the 'To age'. This causes a problem when the default value for 'To age' is pre-filled with 0Y 2M.

Considering the design, the user is lead to start filling the form by 'From age', but with this restriction, for now the user needs to edit 'To age' first.

For example, if the interval is supposed to be "From age" = 5Y ", To age"= 10Y, the only way to make it work is inputting 10Y first. See video below:

https://github.com/user-attachments/assets/32bea7b8-08b4-48a7-bdca-b9437d2b0194

If not done in this order, the system will not accept the number 5Y, as the default value for 'To age' is 0Y 2M
![Screenshot 2024-09-26 at 4 19 49 PM](https://github.com/user-attachments/assets/63152c6c-a792-4132-952f-ccffbc6d9409)

## Expected behaviour 🤔

If 'To age' < ' From age', error message appears.

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Go to 'Programs' >Immunisations
2. Click on 'Add Vaccine course'
3. Scroll down to '+ Dose'
4. Try typing 'From age' = 3Y 
5. See error

## Your environment 🌱

- Open mSupply Version: V2.3.RC1
- Legacy mSupply Central Server Version: V7.17

- Platform: Desktop app

**Comment by lache-melvin** [comment]
Yep agree this is pretty unusable - would appear the usability is at least somewhat constrained by the points outlined in #4874, solution for now can be to remove any limits on the inputs while typing, and validate on save 👍

**Comment by CarlosNZ** [comment]
I would just add, when you add a new dose, the default minimum is going to one month after the previous minimum, whereas if should be whatever the previous maximum was. Just noting so we can fix here as well.

**Comment by lache-melvin** [comment]
Closed by #4931

**Comment by marthakeezy** [comment]
# 🧪 Testing



`V2.6.0-build-test-3`

- [ ] Can change min age value without being constrained by the current max
- [ ] Ranges can overlap (obviously this is not what we want, but solution to that will be done elsewhere)

**Comment by marthakeezy** [comment]
hey team - @lache-melvin & @CarlosNZ 

not sure if this is the same issue, but i seem to be able to set higher 'from age' than 'to age'

but according to the original issue it shouldn't be possible? or am i misreading?

**Comment by lache-melvin** [comment]
Ah great catch @marthakeezy! Looks like in this issue it was intended to remove the limits while inputting, but validate them on save, but the validation is clearly missing!

I don't think this breaks anything - just that the vaccine card would just show those vaccines as overdue immediately after they become available. Which, not ideal, but not critical. 

Will reopen and hopefully be able to get to within the vax card changes this release 🙏

**Comment by marthakeezy** [comment]
nice thanks!!

**Comment by Chris-Petty** [comment]
Fixed in 2.6.0-RC5

**Comment by marthakeezy** [comment]
hey team!

1. should i be able to add decimal values for months?

<img width="956" alt="Image" src="https://github.com/user-attachments/assets/2098135f-eac9-4f3f-a328-e7f96c774df2" />

2. should it be possible for the doses to overlap with their ages? e.g. here a dose is until 45 y.o. and the next one starts at 40 y.o.

<img width="924" alt="Image" src="https://github.com/user-attachments/assets/1b63806c-abe0-4d3a-b0c6-89d9491241b2" />

3. if i set a minimum interval (in days) to 60, why am i able to save the age of the next dose from 1M to 2M which precedes the interval mark? 

<img width="938" alt="Image" src="https://github.com/user-attachments/assets/2d71e6b1-ec31-4c34-b160-0d52562c8804" />

**Comment by lache-melvin** [comment]
> 1. should i be able to add decimal values for months?

Yep, e.g. for 1.5 months = 6 weeks 

> 2. should it be possible for the doses to overlap with their ages? e.g. here a dose is until 45 y.o. and the next one starts at 40 y.o.

Yep, if you can get a dose at any age, then max age would be 99+, but still need to be able to set appropriate minimum ages for subsequent doses

> 3. if i set a minimum interval (in days) to 60, why am i able to save the age of the next dose from 1M to 2M which precedes the interval mark?

Lol this one probably doesn't make the most sense, but I don't really think it's a bug. These time frames all just control the display of different statuses, suggested dates in the vaccine card... doesn't impact critical logic where we really need to make sure they can't enter anything invalid. 

This is config that only happens on the central server, and is one-time set up for the vaccine course. We trust this administrator to enter sensible inputs for the vaccination program in the country 🙏

**Comment by marthakeezy** [comment]
nice, thanks laché! then QA is done, moving to done ✅
